src/Nevermind/Nevermind.Evm/TransactionStore.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using Nevermind.Core;
+using Nevermind.Core.Crypto;
+
+namespace Nevermind.Evm
+{
+    public class TransactionStore : ITransactionStore
+    {
+        private readonly Dictionary<Keccak, Transaction> _transactions = new Dictionary<Keccak, Transaction>();
+        private readonly Dictionary<Keccak, TransactionReceipt> _transactionRecepits = new Dictionary<Keccak, TransactionReceipt>();
+        private readonly HashSet<Keccak> _processedTransations = new HashSet<Keccak>();
+        private readonly Dictionary<Keccak, Keccak> _blockHashes = new Dictionary<Keccak, Keccak>();
```

Comment:
will be IO later so not critical but less discuss memory implications just to keep that in mind in the future